identity-service/src/notifications/index.js

```diff
@@ -104,8 +63,16 @@ class NotificationProcessor {
       // this.emailQueue.add({ type: 'unreadEmailJob' })
 
       try {
-        // Index notifications
-        let maxBlockNumber = await this.indexNotifications(minBlock)
+        const oldMaxBlockNumber = await this.redis.get('maxBlockNumber')
+        let maxBlockNumber = null
+        // Index notifications and milestones
+        if (minBlock < oldMaxBlockNumber) {
```

Comment:
@hareeshnagaraj i want your thoughts on this logic when you get a chance. sometimes i'm seeing the same minBlock processed over and over and it's sending double push notifications. This should prevent that in a running process. Not sure what's causing it, but this seems to prevent it

Reply:
I think this is explicitly the case per our conversation, but the language / naming here is confusing (even before this PR)

As long as this has no effect on indexing moving fwd, i think it should be all good

Reply:
@hareeshnagaraj i believe this should have no effect moving forward. I see three cases here

1. first time. there's no maxBlockNumber in redis, so it doesn't go into the if statement and indexes because min < null is false
2. minBlock > oldMaxBlockNumber or the two are equal. this should be the happy path where the max block we left off with last time is the min block this time.
3. minBlock < maxBlockNumber. this is basically the only case we're trying to catch and short circuit. We set the value of maxBlockNumber and we don't index so the next time we add a job to the bull queue and check redis, the two values should be equal

Reply:
the returned value from indexAll is the latest max block number returned from the disc prov /notifications route and the minBlockNumber is the highest blockNumber stored in the identity service database, which explains why this condition works...but still misleading